flink/v1.20/flink/src/main/java/org/apache/iceberg/flink/maintenance/stream/TableMaintenance.java

```diff
@@ -0,0 +1,338 @@
+          inputStream == null, "Can't set maxReadBack when change stream is provided");
+      this.maxReadBack = newMaxReadBack;
+      return this;
+    }
+
+    /**
+     * Adds a specific task with the given schedule.
+     *
+     * @param task to add
+     * @return for chained calls
+     */
+    public Builder add(MaintenanceTaskBuilder<?> task) {
+      taskBuilders.add(task);
+      return this;
+    }
+
+    /** Builds the task graph for the maintenance tasks. */
+    public void append() {
+      Preconditions.checkArgument(!taskBuilders.isEmpty(), "Provide at least one task");
+      Preconditions.checkNotNull(uidSuffix, "Uid suffix should no be null");
+
+      List<String> taskNames = Lists.newArrayListWithCapacity(taskBuilders.size());
+      List<TriggerEvaluator> evaluators = Lists.newArrayListWithCapacity(taskBuilders.size());
+      for (int i = 0; i < taskBuilders.size(); ++i) {
+        taskNames.add(nameFor(taskBuilders.get(i), i));
+        evaluators.add(taskBuilders.get(i).evaluator());
+      }
+
+      DataStream<Trigger> triggers =
+          DataStreamUtils.reinterpretAsKeyedStream(changeStream(), unused -> true)
+              .process(
+                  new TriggerManager(
+                      tableLoader,
+                      lockFactory,
+                      taskNames,
+                      evaluators,
+                      rateLimit.toMillis(),
+                      lockCheckDelay.toMillis()))
+              .name(TRIGGER_MANAGER_TASK_NAME)
+              .uid("trigger-manager-" + uidSuffix)
+              .slotSharingGroup(slotSharingGroup)
+              .forceNonParallel()
+              .assignTimestampsAndWatermarks(new PunctuatedWatermarkStrategy())
+              .name("Watermark Assigner")
+              .uid("watermark-assigner-" + uidSuffix)
+              .slotSharingGroup(slotSharingGroup)
+              .forceNonParallel();
+
+      // Add the specific tasks
+      DataStream<TaskResult> unioned = null;
+      for (int i = 0; i < taskBuilders.size(); ++i) {
+        int finalIndex = i;
+        DataStream<Trigger> filtered =
+            triggers
+                .filter(t -> t.taskId() != null && t.taskId() == finalIndex)
+                .name("Filter " + i)
+                .forceNonParallel()
+                .uid("filter-" + i + "-" + uidSuffix)
+                .slotSharingGroup(slotSharingGroup);
+        MaintenanceTaskBuilder<?> builder = taskBuilders.get(i);
+        DataStream<TaskResult> result =
+            builder.append(
+                filtered,
+                i,
+                taskNames.get(i),
+                tableLoader,
+                uidSuffix,
+                slotSharingGroup,
+                parallelism);
+        if (unioned == null) {
+          unioned = result;
+        } else {
+          unioned = unioned.union(result);
+        }
+      }
+
+      // Add the LockRemover to the end
+      unioned
+          .global()
```

Comment:
do we need `global` with the `forceNonParallel` below?

Reply:
removed the global